test/integration/gardenlet/bastion/bastion_test.go

```diff
@@ -0,0 +1,290 @@
+				Ingress: []operationsv1alpha1.BastionIngressPolicy{{
+					IPBlock: networkingv1.IPBlock{CIDR: "1.2.3.4/32"},
+				}},
+			},
+		}
+	})
+
+	JustBeforeEach(func() {
+		By("Create Seed")
+		Expect(testClient.Create(ctx, seed)).To(Succeed())
+		log.Info("Created Seed for test", "seed", seed.Name)
+
+		DeferCleanup(func() {
+			By("Delete Seed")
+			Expect(testClient.Delete(ctx, seed)).To(Or(Succeed(), BeNotFoundError()))
+		})
+
+		By("Create Shoot")
+		shoot.Spec.SeedName = &seed.Name
+		Expect(testClient.Create(ctx, shoot)).To(Succeed())
+		log.Info("Created Shoot for test", "shoot", client.ObjectKeyFromObject(shoot))
+
+		DeferCleanup(func() {
+			By("Delete Shoot")
+			Expect(testClient.Delete(ctx, shoot)).To(Or(Succeed(), BeNotFoundError()))
+		})
+
+		By("Wait until manager has observed shoot")
+		Eventually(func() error {
+			return mgrClient.Get(ctx, client.ObjectKeyFromObject(shoot), &gardencorev1beta1.Shoot{})
+		}).Should(Succeed())
+
+		By("Patch the Shoot status with TechincalID")
+		patch := client.MergeFrom(shoot.DeepCopy())
+		projectName := strings.TrimPrefix(testNamespace.Name, "garden-")
+		shootTechincalId := shootpkg.ComputeTechnicalID(projectName, shoot)
+		shoot.Status.TechnicalID = shootTechincalId
+		Expect(testClient.Status().Patch(ctx, shoot, patch)).To(Succeed())
+
+		By("creating seed namespace for test")
+		seedNamespace = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: shootTechincalId,
+			},
+		}
+
+		Expect(testClient.Create(ctx, seedNamespace)).To(Succeed())
+		log.Info("Created seed Namespace for test", "namespaceName", seedNamespace.Name)
+
+		DeferCleanup(func() {
+			By("deleting seed namespace")
+			Expect(testClient.Delete(ctx, seedNamespace)).To(Or(Succeed(), BeNotFoundError()))
+		})
+
+		By("Create Bastion")
+		operationsBastion.Spec.ShootRef = corev1.LocalObjectReference{Name: shoot.Name}
+		operationsBastion.Spec.SeedName = &seed.Name
+		Expect(testClient.Create(ctx, operationsBastion)).To(Succeed())
+		log.Info("Created Bastion for test", "bastion", client.ObjectKeyFromObject(operationsBastion))
+
+		extensionsBastion = &extensionsv1alpha1.Bastion{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      operationsBastion.Name,
+				Namespace: seedNamespace.Name,
+			},
+		}
+
+		DeferCleanup(func() {
+			By("Delete Bastion")
+			Expect(testClient.Delete(ctx, operationsBastion)).To(Or(Succeed(), BeNotFoundError()))
+
+			By("Wait for extension Bastion to be gone")
+			Eventually(func() error {
+				return mgrClient.Get(ctx, client.ObjectKeyFromObject(extensionsBastion), extensionsBastion)
+			}).Should(BeNotFoundError())
```

Comment:
This should be asserted in the `It`. We are not creating this resource explicitly.